src/templates/partials/iis.hbs

```diff
@@ -0,0 +1,187 @@
+}
+
+function Reset-KeyExchangeAlgorithms(){
+  Write-Host "Reset-KeyExchangeAlgorithms"
+  $keyExchangeAlgorithmsKey = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\KeyExchangeAlgorithms"
+  Remove-Item -Path $keyExchangeAlgorithmsKey -Recurse -Force
+  $return = New-Item -Path $keyExchangeAlgorithmsKey
+}
+
+function Set-SChannelProtocol($name, $value, $clientAlso){
+  Write-Host "Set-SChannelProtocol ``$name`` ``$value``"
+  $protocolKey = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$name"
+  if($false -eq (Test-Path $protocolKey)){
+    $return = New-Item -Path $protocolKey
+  }
+  $protocolKey = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$name\Server"
+  if($false -eq (Test-Path $protocolKey)){
+    $return = New-Item -Path $protocolKey
+  }
+  Set-ItemProperty -Path $protocolKey -Name "Enabled" -Value $value -Type DWord
+  if($value -eq 0x00000000){
+    $return = Set-ItemProperty -Path $protocolKey -Name "DisabledByDefault" -Value 0x00000001 -Type DWord
+  }else{
+    $return = Set-ItemProperty -Path $protocolKey -Name "DisabledByDefault" -Value 0x00000000 -Type DWord
+  }
+  if($clientAlso){
+    $protocolKey = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$name\Client"
+    if($false -eq (Test-Path $protocolKey)){
+      $return = New-Item -Path $protocolKey
+    }
+    if($value -eq 0x00000000){
+      $return = Set-ItemProperty -Path $protocolKey -Name "DisabledByDefault" -Value 0x00000001 -Type DWord
+    }else{
+      $return = Set-ItemProperty -Path $protocolKey -Name "DisabledByDefault" -Value 0x00000000 -Type DWord
+    }
+  }
+}
+
+function Reset-SChannelProtocols(){
+  Write-Host "Reset-SChannelProtocols"
+  $protocolKey = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\"
+  $r = Remove-Item -Path $protocolKey -Recurse -Force
+  $return = New-Item -Path $protocolKey
+}
+
+function Remove-RegistryValue($path, $name){
+  Write-Host "Remove-RegistryValue ``$name``"
+  $key = Get-Item -LiteralPath $path
+  if ($key.GetValue($name, $null) -ne $null) {
+    $r = Remove-ItemProperty -Path $path -Name $name
+  }
+}
+
+# Restrict the use of certain cryptographic algorithms and protocols in Schannel.dll
+# http://support2.microsoft.com/default.aspx?scid=kb;EN-US;245030
+Reset-Chiphers
+Set-Chipher -name "AES 128/128" -value 0xffffffff
+Set-Chipher -name "AES 256/256" -value 0xffffffff
+Set-Chipher -name "DES 56/56" -value 0x00000000
+Set-Chipher -name "NULL" -value 0x00000000
+Set-Chipher -name "RC2 128/128" -value 0x00000000
+Set-Chipher -name "RC2 40/128" -value 0x00000000
+Set-Chipher -name "RC2 56/128" -value 0x00000000
+Set-Chipher -name "RC4 128/128" -value 0x00000000
+Set-Chipher -name "RC4 40/128" -value 0x00000000
+Set-Chipher -name "RC4 56/128" -value 0x00000000
+Set-Chipher -name "RC4 64/128" -value 0x00000000
+Set-Chipher -name "Triple DES 168/168" -value 0x00000000
```

Comment:
Is there no way to simply create a whitelist? This seems really inelegant.

Reply:
I could write a `Disable-Cipher`/`Enable-Cipher` method that takes an array `@("RC4 40/128", "RC4 56/128", ...)` as input...

Reply:
I don't know, is there not a way to disable all ciphers and then enable them one by one? I'm not particularly familiar with IIS.

Reply:
The values are keys in the registry. They are present and enabled by default :-(. I could write a method which iterates the keys and disables each without naming/listing each one. The a second method to enable the preferred ones. would this be fine?

Reply:
that looks like a list of algorithms supported on the system level, not specifically for TLS...
are you sure that they need to be disabled for the TLS ciphers to be disabled?
I don't think we want to break backwards compatibility with PKCS#12 files if the user configures their stack TLS properly.

Reply:
It is the configuration for [SChannel](https://docs.microsoft.com/en-us/windows/win32/secauthn/secure-channel) which is used for encrypted communication. 

> Microsoft TLS/SSL Security Provider, the Schannel.dll file, uses the CSPs that are listed here to conduct secure communications over SSL or TLS in its support for Internet Explorer and Internet Information Services (IIS). 
[Source](https://support.microsoft.com/en-us/help/245030/how-to-restrict-the-use-of-certain-cryptographic-algorithms-and-protoc)

This will also affect apps running on the platform - e.g. PowerShell, Outlook and so on **when communication encrypted over the wire**.

What do you mean by *backwards compatibility with PKCS#12 files*?

Reply:
> What do you mean by backwards compatibility with PKCS#12 files?

I was thinking of ciphers available in TLS 1.0, where a 128 bit RC2 was never defined, but I forgot about SSLv2 ciphers. At the same time, there are RC2 ciphers defined for use in PKCS#12 files and they have the same key sizes as SSLv2 ciphers.

Reply:
SChannel has nothing to do with the crypto operations on files.